### ISSUE TITLE
One example in wiki doesn't seem to work

### ISSUE THREAD

**Issue by hansonchar** [stedolan/jq#2557]
**Describe the bug**
One example at the [Path Expression wiki](https://github.com/stedolan/jq/wiki/jq-Language-Description#Path-Expressions) doesn't seem to work.  In particular,
> Given a datum like `{"a":{"b":[{"c":0},{"d":1}]}}` we can have path expressions like:
> `.a[]b[][]` => matches all the paths in this input

However, 

```
$ echo '{"a":{"b":[{"c":0},{"d":1}]}}' | jq -c '.a[]b[][]'
jq: error: syntax error, unexpected IDENT, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
.a[]b[][]
jq: 1 compile error
```

A contradiction?

**Environment (please complete the following information):**
 - OS and Version: [e.g. macOS, Windows, Linux (please specify distro)]
 - jq version [e.g. 1.5]
 
```
$ uname -a
Darwin 88665a29535f.ant.amazon.com 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 19 20:44:01 PST 2022; root:xnu-8020.240.18~2/RELEASE_X86_64 x86_64

$ jq --version
jq-1.6
```

**Comment by wader** [comment]
Yes looks weird, was it suppose to say `.a[][][]`? also "matches all the paths in this input" is a bit misleading, sounds like it would behave as `..`? maybe something like "matches all leaf paths"?

**Comment by wader** [comment]
👍
